app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -40,4 +41,8 @@ def validate_default_value
   def puppet?
     true
   end
+
+  def check_override_selected
+    errors.add(:override, _("must be true to edit the parameter")) unless self.override || (self.changes.size == 1 && self.changes["description"].present?) || self.new_record?
```

Comment:
So this condition can be reduced to `if override && persisted?`? If so, can you move that to the `before_validation` definition in line 5 so we don't even call the method?